mealy/error_visualizer.py

```diff
@@ -199,55 +225,54 @@ def plot_feature_distributions_on_leaves(self, leaf_selector='all_errors',
             show_class (bool): show the proportion of Wrongly and Correctly predicted samples in the feature
                 distributions.
 
-            rank_leaves_by (str): ranking criterium for the leaf nodes. It can be either 'purity' to rank by the leaf
-                node purity (ratio of wrongly predicted samples over the total for an error node) or 'class_difference'
-                (difference of number of wrongly and correctly predicted samples in a node).
+            rank_leaves_by (str): ranking criterion for the leaf nodes. Valid values are:
+                * 'total_error_fraction': rank by the fraction of total error in the node
+                * 'purity': rank by the purity (ratio of wrongly predicted samples over the total number of node samples)
+                * 'class_difference': rank by the difference of number of wrongly and correctly predicted samples
+                in a node.
 
             nr_bins (int): number of bins in the feature distribution plots.
 
             figsize (tuple): size of the plots.
 
         """
 
-        error_class_idx = np.where(self._error_clf.classes_ == ErrorAnalyzerConstants.WRONG_PREDICTION)[0][0]
-        correct_class_idx = 1 - error_class_idx
+        correct_class_idx = 1 - self._error_tree.error_class_idx
 
         ranked_feature_ids = rank_features_by_error_correlation(self._error_clf.feature_importances_)
-        if self.pipeline_preprocessor is None:
+        if self._error_analyzer.pipeline_preprocessor is None:
             if top_k_features > 0:
                 ranked_feature_ids = ranked_feature_ids[:top_k_features]
 
-            x, y = self._error_train_x[:, ranked_feature_ids], self._error_train_y
+            x, y = self._error_analyzer._error_train_x[:, ranked_feature_ids], self._error_analyzer._error_train_y
             min_values, max_values = x.min(axis=0), x.max(axis=0)
-            feature_names = self.mpp_feature_names
+            feature_names = self._error_analyzer.preprocessed_feature_names
         else:
-            ranked_feature_ids = [self.pipeline_preprocessor.inverse_transform_feature_id(idx) for idx in
-                                  ranked_feature_ids]
+            ranked_feature_ids = list(set(self._error_analyzer.pipeline_preprocessor.inverse_transform_feature_id(idx) for idx in
+                                          ranked_feature_ids))
```

Comment:
As I briefly mentioned in the last meeting, this is actually very wrong. But I will fix it in another PR :)